src/mps/mps.jl

```diff
@@ -585,9 +585,44 @@ psi = randomMPS(s, n->isodd(n) ? "Up" : "Dn"; linkdims=m)
 Cuu = correlation_matrix(psi,"Cdagup","Cup";site_range=2:8)
 ```
 """
-function correlation_matrix(psi::MPS, Op1::AbstractString, Op2::AbstractString; kwargs...)
+function correlation_matrix(psi::MPS, _Op1::AbstractString, _Op2::AbstractString; kwargs...)
   N = length(psi)
   ElT = promote_itensor_eltype(psi)
+  s = siteinds(psi)
+
+  Op1=_Op1 #make copies into which we can insert "F" string operators.
+  Op2=_Op2
+  onsiteOp = "$Op1*$Op2"
+  fermionic1 = has_fermion_string(Op1, s[1])
+  fermionic2 = has_fermion_string(Op2, s[1])
+  if fermionic1!=fermionic2
+    error("correlation_matrix: Mixed fermionic and bosonic operators are not supported yet.")
+  end
+  
+  # Decide if we need to calculate a non-hermitian corr. matrix which is roughly double the work.
+  isHermitian=false #Assume corr-matrix is non-hermitian
+  if haskey(kwargs, :isHermitian) #Did the user explicitly request something?
```

Comment:
Small style change: for user-facing code (like function names, keyword arguments, etc.) we generally try to follow the Julia naming convention of using all lower case. In this case, I would recommend `ishermitian`. For things that are hard to read in that style, you can use underscores (like `has_fermion_string`).

It clashes with the C++ style we use so can be tough switching back and forth, but in general it makes it much easier to follow the Julia style since we often overload Julia functions in generic code.